packages/@webex/plugin-cc/src/cc.ts

```
@@ -873,10 +878,109 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
    * messages, and client-side events, then securely submits them to Webex's diagnostics
    * service. The returned tracking ID, feedbackID can be provided to Webex support for faster
    * issue resolution.
-   * @returns Promise<SubmitLogsResponse>
+   * @returns Promise<UploadLogsResponse>
    * @throws Error
    */
   public async uploadLogs(): Promise<UploadLogsResponse> {
     return this.webexRequest.uploadLogs();
   }
+
+  /**
+   * Updates the agent device type.
+   * This method allows the agent to change their device type (e.g., from BROWSER to EXTENSION or anything else).
+   * It will also throw an error if the new device type is the same as the current one.
+   * @param data type is AgentDeviceUpdate - The data required to update the agent device type, including the new login option and dial number.
+   * @returns Promise<UpdateDeviceTypeResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * const data = {
+   *   loginOption: 'EXTENSION',
+   *   dialNumber: '1234567890',
+   * };
+   * const result = await webex.cc.updateAgentDeviceType(data);
+   * ```
+   */
+  public async updateAgentDeviceType(data: AgentDeviceUpdate): Promise<UpdateDeviceTypeResponse> {
+    this.metricsManager.timeEvent([
+      METRIC_EVENT_NAMES.AGENT_DEVICE_TYPE_UPDATE_SUCCESS,
+      METRIC_EVENT_NAMES.AGENT_DEVICE_TYPE_UPDATE_FAILED,
+    ]);
+
+    LoggerProxy.info('updateAgentDeviceType | starting profile update', {
+      module: CC_FILE,
+      method: this.updateAgentDeviceType.name,
+    });
+
+    try {
+      // ensure we change device type
+      if (this.webCallingService?.loginOption === data.loginOption) {
+        const message = 'New Device type is same as current device type';
```

Also this is something I observed in Agent desktop that it doesn't let you submit the selection if it's the same device type so throwing error for the same, is it the right way to handle it. Or if you wanna throw the error the maybe in the message we should not say it failed but something like we won;t proceed with updating device type as there is no change in it

---

That would be something to be handled on the widgets, yes.
I can do one thing I want to throw the error just so developers who are experimenting with it can identify - the other details can be taken care of in the app consuming the SDK.
I can modify the message to - "Will not proceed with device update as new Device type is same as current device type"